### ISSUE TITLE
Display User Reconnection on Admin Frontend

### ISSUE THREAD

**Issue by ajohns619** [JumboCode/E4P#199]
Conversations in the persist_conversation branch stick around for some time after the user leaves to allow for re-connections. This means the chat should no longer be deactivated immediately when the user leaves. Instead, the admin should receive a notice that the user has left, and receive another notice when the user returns. If the user does not return and that conversation is removed from the server's array, the admin front-end should be notified, so the admin has the option to delete this chat.

**Comment by ajohns619** [comment]
Once #179 is done, we can merge master into this branch and display these notifications as updates to the user. Before persist_conversation can be merged into master, the "disconnect" button on the index page needs to be removed, as well as the "disconnect" function in index.js.

**Comment by ajohns619** [comment]
On the user's side, they should get a notice if they reconnect unsuccessfully and should not be able to type any more messages.

**Comment by ajohns619** [comment]
On the server side, removeConversation should remove all sockets currently in that conversation's room. This way, the admin doesn't stick around in old rooms. (https://stackoverflow.com/questions/23342395/how-to-delete-a-room-in-socket-io)

**Comment by StevieSong** [comment]
user disconnect/reconnect was sending 'user matched' before 'user reconnect' because of old logic to select for user's room. this is now outdated since we allow users to reconnect with new id's that aren't their room explicitly. refactor server to not send extraneous messages

**Comment by StevieSong** [comment]
> On the server side, removeConversation should remove all sockets currently in that conversation's room. This way, the admin doesn't stick around in old rooms. (https://stackoverflow.com/questions/23342395/how-to-delete-a-room-in-socket-io)

for reference, the above solution only applies to socket versions < 1.0
https://stackoverflow.com/questions/6563885/socket-io-how-do-i-get-a-list-of-connected-sockets-clients
https://stackoverflow.com/questions/9352549/getting-how-many-people-are-in-a-chat-room-in-socket-io#24425207

**Comment by ajohns619** [comment]
Merged: https://github.com/JumboCode/E4P/commit/da24dad8483a9025302cfc4c4285599641d6d83b
